### ISSUE TITLE
Gateway position unexpected layout 

### ISSUE THREAD

**Issue by abdul99ahad** [bpmn-io/bpmn-auto-layout#67]
### Describe the Bug

Sorting the elements to postpone laying out the gateways works in some cases but here's another case where it leads to new unexpected behavior:

![image](https://github.com/user-attachments/assets/5b60d366-65e9-491e-895f-fef189fb58b0)

The gateway is postponed so it's laid out only once all elements on the bottom have been laid out.

Even worse in this case (but was also bad before):

![image](https://github.com/user-attachments/assets/f35b7e32-55a1-45e8-87f5-876117ce3cbc)

### Steps to Reproduce

1. do this
2. do that
3. now this happens

### Expected Behavior
The model should adapt some different strategy to model gateways and elements as we can't rely on sorting for every scenario.

### Environment

- Host (Browser/Node version), if applicable: [e.g. MS Edge 18, Chrome 69, Node 10 LTS]
- OS: [e.g. Windows 7]
- Library version: [e.g. 2.0.0]
